src/oic/oic/consumer.py

```diff
@@ -461,3 +515,45 @@ def check_session(self):
 
     def end_session(self):
         pass
+
+    # LOGOUT related
+
+    def backchannel_logout(self, request=None, request_args=None):
+        """
+        Receives a back channel logout request and returns a Session ID
+        if the request was OK.
+
+        :param request: A urlencoded request
+        :param request_args: The request as a dictionary
+        :return: A Session Identifier
+        """
+        if request:
+            req = BackChannelLogoutRequest().from_urlencoded(as_unicode(request))
+        else:
+            req = BackChannelLogoutRequest(**request_args)
+
+        kwargs = {"aud": self.client_id, "iss": self.issuer, "keyjar": self.keyjar}
+
+        try:
+            req.verify(**kwargs)
+        except (MessageException, ValueError, NotForMe) as err:
+            raise MessageException("Bogus logout request: {}".format(err))
+
+        # Find the subject through 'sid' or 'sub'
+
+        try:
+            sub = req["logout_token"]["sub"]
+        except KeyError:
+            try:
+                sm_id = req["logout_token"]["sid"]
+            except KeyError:
+                raise MessageException('Neither "sid" nor "sub"')
+            else:
+                _sid = session_get(self.sso_db, "smid", sm_id)
```

Comment:
Since this is in a separate database, we can probably still call it `sid` instead of `smid`. This would reduce the number of methods that need to be implemented.

And in reality you probably really want two separate storages, so there is little risk of people targeting single storage with both actions.

Reply:
True

Reply:
But I'd keep smidges since sid used for the key to the database. Would be confusing if sid was both the key and a property in the DB.
Session ID (sid) and Session Management ID (smid) are different things.

Reply:
OK, we can keep it that way.